docking.js

```diff
@@ -1652,32 +1687,163 @@ const DockedDash = new Lang.Class({
     },
 
     _showOverlay: function() {
-        if (this._settings.get_boolean('hotkeys-overlay'))
-            this.dash.toggleNumberOverlay(true);
-
-        // Restart the counting if the shortcut is pressed again
-        if (this._numberOverlayTimeoutId) {
-            Mainloop.source_remove(this._numberOverlayTimeoutId);
-            this._numberOverlayTimeoutId = 0;
-        }
+        for (let i = 0; i < allDocks.length; i ++) {
```

Comment:
Agreed, this part is simply horrible, I'll try to rethink this part. Moving it to the manager level seems like a good idea, let's see how the implementation actually looks.

As for the multiple screens: yes, it's actually quite nice, applications will be launched depending on what monitor the mouse is.